File: Tienda/src/app/model/Producto.ts

export class Producto {
  private _id: number;
  private _nombre: string;
  private _seccion: string;
  private _precio: number;
  private _stock: number;

  constructor(id: number, nombre: string, seccion: string, precio: number, stock:number) {
    this._id = id;
    this._nombre = nombre;
    this._seccion = seccion;
    this._precio = precio;
    this._stock = stock;
  }

  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get nombre(): string {
    return this._nombre;
  }
  public set nombre(value: string) {
    this._nombre = value;
  }

  public get seccion(): string {
    return this._seccion;
  }
  public set seccion(value: string) {
    this._seccion = value;
  }

  public get precio(): number {
    return this._precio;
  }
  public set precio(value: number) {
    this._precio = value;
  }

  public get stock(): number {
    return this._stock;
  }
  public set stock(value: number) {
    this._stock = value;
  }
}

File: 08_tienda_persistencia_yo/src/main/java/model/Producto.java
package model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name="productos")
@NamedQuery(name = "Producto.readBySection", query = "select p from Producto p where p.seccion = :seccion")
@NamedQuery(name = "Producto.readByName", query = "select p from Producto p where p.nombre = :nombre")
public class Producto {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int idProducto;
	private String nombre;
	private String seccion;
	private double precio;
	private int stock;
}
